dashboard/src/components/SideMenu/SideMenu.tsx

```diff
@@ -65,6 +46,45 @@ const SideMenu = (): JSX.Element => {
   const notSelectedItemClassName =
     'w-full flex pl-5 py-4 cursor-pointer text-white';
 
+  const useNavigateTo = (path: string): (() => void) => {
+    const router = useRouter();
+    const searchParams = useSearch({ strict: false });
+    const org = searchParams.origin;
```

Comment:
using `{ stict: false }` means that it is unsafe and it needs to be validated, otherwise it can break the entire app, you can do something like 

const { origin: unsafeOrigin } = useSearch({ strict: false })
and validate it using zod

`const origin = zOrigin.parse(unsafeOrigin)` 

`zOrigin` is defined in the types folder